<launch>
    <arg name="ns" default="tracking_ros" />
    <arg name="device" default="cuda:0" />
    <arg name="mode" default="default" />
    <arg name="mask" default="false" />

    <arg name="input_image" default="/$(arg ns)/processed_image"/>
    <arg name="output_image" default="/$(arg ns)/segmentation_image"/>
    <arg name="input_segmentation" default="/$(arg ns)/input_segmentation" />
    <arg name="segmentation_image" default="/$(arg ns)/segmentation_image" />
    <arg name="output_segmentation" default="/$(arg ns)/segmentation" />
    <arg name="masked_image" default="/$(arg ns)/masked_image" />

    <node name="track_node"
          pkg="tracking_ros" type="track_node.py"
          output="screen" >
        <remap from="~input_image" to="$(arg input_image)" />
        <remap from="~input_segmentation" to="$(arg input_segmentation)" />
        <remap from="~output_image" to="$(arg segmentation_image)" />
        <remap from="~segmentation" to="$(arg output_segmentation)" />
        <remap from="~masked_image" to="$(arg masked_image)" />
        <rosparam subst_value="true" >
            device: $(arg device)
            mode: $(arg mode)
            model_dir: $(find tracking_ros)/checkpoints/
            tracker_config: $(find tracking_ros)/config/tracker_config.yaml
            del_bg: $(arg mask)
        </rosparam>
    </node>
</launch>
